src/components/Text/Text.tsx

```diff
@@ -1,37 +1,80 @@
 import clsx from 'clsx';
-import React, { ReactNode } from 'react';
+import React from 'react';
 import styles from './Text.module.css';
 
-export interface Props {
-  /**
-   * The rendered tag name of the Heading. The tag name should always relate to the [document outline](http://html5doctor.com/outlines/) and a tag name should never be chosen for its default aesthetic qualities. If a specific style is desired, use the `size` prop to manipulate the Heading style.
-   */
-  as: 'p' | 'span';
-  /**
-   * The child node(s)
-   */
-  children?: ReactNode;
+export type Size =
+  | 'h1'
+  | 'h2'
+  | 'h3'
+  | 'h4'
+  | 'h5'
+  | 'body'
+  | 'sm'
+  | 'xs'
+  | 'caption'
+  | 'overline';
+
+export type Color =
+  | 'alert'
+  | 'base'
+  | 'brand'
+  | 'info'
+  | 'inherit'
+  | 'neutral'
+  | 'success'
+  | 'warning'
+  | 'white';
+
+export type Props = {
   /**
-   * CSS class names that can be appended to the component.
+   * Controls whether to render text inline (defaults to "p");
    */
+  as?: 'p' | 'span';
+  children: React.ReactNode;
   className?: string;
-}
+  color?: Color;
+  size?: Size;
```

Comment:
No `variant` prop as there are a lot of modifying props, should either `size` or `color` (or other prop) be the `variant` prop?

Reply:
@jinlee93 I reckon `color` ought to be `variant`, as `size` is being properly used to control the text size and `color` is the primary stylistic differentiator.

Reply:
I could see `size` becoming `variant` because it actually controls a couple of different style properties (ie text transform and weight) in addition to size.

Reply:
I think other components more commonly control color through `variant`, so renaming `color` makes sense to me. Text is a special case though so I'd be fine keeping it as-is for now and making a note to check once all components are in whether this is the only place we're still using `color` (and if so, should definitely rename it)

Reply:
I'm down to rename it as the color selections themselves sound more "variant-y" anyways